fe/fe-core/src/main/java/com/starrocks/qe/scheduler/ICoordinator.java

```diff
@@ -0,0 +1,175 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.qe.scheduler;
+
+import com.starrocks.analysis.DescriptorTable;
+import com.starrocks.common.Status;
+import com.starrocks.common.util.RuntimeProfile;
+import com.starrocks.planner.PlanFragment;
+import com.starrocks.planner.ScanNode;
+import com.starrocks.planner.StreamLoadPlanner;
+import com.starrocks.proto.PPlanFragmentCancelReason;
+import com.starrocks.proto.PQueryStatistics;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.QueryStatisticsItem;
+import com.starrocks.qe.RowBatch;
+import com.starrocks.sql.LoadPlanner;
+import com.starrocks.thrift.TDescriptorTable;
+import com.starrocks.thrift.TLoadJobType;
+import com.starrocks.thrift.TNetworkAddress;
+import com.starrocks.thrift.TReportExecStatusParams;
+import com.starrocks.thrift.TSinkCommitInfo;
+import com.starrocks.thrift.TTabletCommitInfo;
+import com.starrocks.thrift.TTabletFailInfo;
+import com.starrocks.thrift.TUniqueId;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public interface ICoordinator {
```

Comment:
```suggestion
public interface Coordinator {
```

Reply:
I don't rename `Coordinator` to `DefaultCoordinator` in this PR. Otherwise, the git diff isn't able to track the line modification of `Coordinator`.

Therefore, the interface cannot use the name `Coordinator` for now. I would rename  `Coordinator` to `DefaultCoordinator`, `ICoordinator` to `Coordinator` in the future PR.

Reply:
BTW, I think the class better to be an abstract class, not interface

Reply:
I just think the method of `interface` should be more concise and closer to the behavior, instead of deal with `scannode`, `profile`, `url`... these methods are related to implementation

Reply:
Sounds reasonable. Let me modify it.

Reply:
Rename rename `Coordinator` to `DefaultCoordinator`, `ICoordinator` to `Coordinator`